src/managers/slidedownManager/SlidedownManager.ts

```diff
@@ -119,19 +123,19 @@ export class SlidedownManager {
     }
 
     const email = ChannelCaptureContainer.getValueFromEmailInput();
-    this.context.updateManager.updateEmail(email);
+    this.updateEmail(email);
   }
 
   private async handleAllowForSmsType(): Promise<void> {
     const smsInputFieldIsValid = OneSignal.slidedown.channelCaptureContainer.smsInputFieldIsValid;
     const isSmsEmpty = ChannelCaptureContainer.isSmsInputFieldEmpty();
+    const sms = ChannelCaptureContainer.getValueFromSmsInput();
 
-    if (!smsInputFieldIsValid || isSmsEmpty) {
+    if (!sms || !smsInputFieldIsValid || isSmsEmpty) {
```

Comment:
this is redundant since we already check for `isSmsEmpty`

the function `isSmsInputFieldEmpty` will return the value or an empty string